app/controller/button/TracingMixin.js

```diff
@@ -0,0 +1,275 @@
+/**
+ * A mixin to add tracing functionality to a drawing tool.
+ *
+ * The basic functionality is taken from the official OpenLayers example:
+ * https://openlayers.org/en/latest/examples/tracing.html
+ *
+ * @class CpsiMapview.controller.button.TracingMixin
+ */
+Ext.define('CpsiMapview.controller.button.TracingMixin', {
+    extend: 'Ext.Mixin',
+
+    requires: [
+        'BasiGX.util.Map',
+        'CpsiMapview.util.Tracing'
+    ],
+
+    /**
+     * The style of the preview line during tracing.
+     */
+    previewStyle: new ol.style.Style({
+        stroke: new ol.style.Stroke({
+            color: 'rgba(255, 0, 0, 1)',
+            width: 10,
+        })
+    }),
+
+    /**
+     * Enhances drawing functionality by adding tracing to it.
+     *
+     * Needs to have access to the variable 'editingIsActive' of the parent component.
+     *
+     * @param {String[]} tracingLayerKeys The keys of the layers to trace
+     */
+    initTracing: function (tracingLayerKeys) {
+        var me = this;
+
+        if (me.getView()) {
+            me.getView().fireEvent('tracingstart');
+        }
+
+        // get tracing layers
+        me.tracingLayers = [];
+        Ext.each(tracingLayerKeys, function (key) {
+            var foundLayers = BasiGX.util.Layer.getLayersBy('layerKey', key);
+            if (foundLayers.length === 1) {
+                me.tracingLayers.push(foundLayers[0]);
+            }
+        });
+
+        // the options for the eachFeature functions
+        me.forEachFeatureOptions = {
+            hitTolerance: 10,
+            layerFilter: function (layer) {
+                return Ext.Array.contains(me.tracingLayers, layer);
+            },
+        };
+
+        me.tracingFeature = new ol.Feature({
+            geometry: new ol.geom.LineString([]),
+        });
+        me.tracingFeatureArray = [];
+        me.tracingStartPoint = null;
+        me.tracingEndPoint = null;
+
+
+        // For debugging the line where the user can trace to can be displayed.
+        var debug = false;
+        if (debug) {
```

Comment:
I'm not sure, if we should keep the code as is. Maybe better provide a proper config variable to the mixin/component that can be switched to true/false.

Reply:
changed to config with 476398f